src/node_url.cc

```diff
@@ -407,29 +407,64 @@ int64_t ParseIPv4Number(const char* start, const char* end) {
   return strtoll(start, nullptr, R);
 }
 
+// https://url.spec.whatwg.org/#ipv4-number-parser
+bool IsIPv4NumberValid(const std::string_view input) {
+  if (input.empty()) {
+    return false;
+  }
+
+  // If a number starts with '0' it might be a number with base 8 or base
+  // 16. If not, checking if all characters are digits proves that it is a
+  // base 10 number.
+  if (input.size() >= 2 && input[0] == '0') {
+    if (input[1] == 'X' || input[1] == 'x') {
+      if (input.size() == 2) {
+        return true;
+      }
+
+      return input.find_first_not_of("0123456789abcdefABCDEF", 2) ==
+             std::string_view::npos;
+    }
+
+    return input.find_first_not_of("01234567", 1) == std::string_view::npos;
+  }
+
+  return std::all_of(input.begin(), input.end(), ::isdigit);
```

Comment:
isdigit is locale-sensitive. This might behave differently depending on the active locale.

Reply:
You are totally right @bnoordhuis. 

```suggestion
  return std::all_of(input.begin(), input.end(), [](char c) { return c >= '0' && c <= '9'; });
```

Reply:
there is a IsASCIIDigit available in this file.. I guess I can use it